src/System.IO.FileSystem/tests/FileStream/WriteAsync.cs

```diff
@@ -418,7 +418,7 @@ public Task ManyConcurrentWriteAsyncs()
                         }
                     } while (!tokenSource.Token.IsCancellationRequested);
                 }
-            } while (DateTime.UtcNow - testStartTime <= testRunTime);
+            } while (DateTime.UtcNow - testStartTime <= testRunTime && totalBytesWritten < 10000000);
```

Comment:
Doesn't this sort of defeat the purpose of the test?  How long does the test take now with this limitation?